asklib.module

```diff
@@ -124,6 +124,47 @@ function asklib_cron() {
       $question->save();
     }
   }
+
+  // Clear contact information of questions older than 2 years every day after 1 am.
+  $request_time = \Drupal::time()->getRequestTime();
+  $next_execution = \Drupal::state()->get('asklib.next_question_contact_cleanup', 0);
+
+  if($request_time > $next_execution) {
+    \Drupal::state()->set('asklib.next_question_contact_cleanup',
+    mktime(1, 0, 0, date("m")  , date("d") + 1, date("Y")));
+
+    _asklib_clear_old_question_contact_info();
+  }
+  
+}
+
+/**
+ * Clears contact information from questions, that are more than two years old.
+ * This is done in small batches to avoid slowing down the server.
+ */
+function _asklib_clear_old_question_contact_info()
+{
+  $two_years_ago = mktime(1, 0, 0, date("m"), date("d"), date("Y") - 2);
+  $storage = \Drupal::entityTypeManager()->getStorage('asklib_question');
+  $query = $storage->getQuery();
+  $orGroup = $query->orConditionGroup()
+    ->exists('email')
+    ->condition('name', 'Kysyjätieto poistettu', '<>');
+  $qids = $query
+    ->condition('created', $two_years_ago, '<=')
+    ->condition($orGroup)
+    ->range(0, 1000)
+    ->execute();
+  
+  if(!empty($qids)) {
+    $query = \Drupal::database()->update('asklib_questions')
+    ->fields([
+      'name' => 'Kysyjätieto poistettu',
```

Comment:
Is this fine to do just with value in finnish?

Reply:
Well the 'name' field itself is not shown anywhere (not even on the admin side). The field is no longer in use and is retained for archival purposes?

I quickly looked up and it seems that the last time this field was in use was in November 2017. Since we are erasing all asker contact info that is older than 2 years, it kinda seems strange to even have this field in the first place.

Reply:
Probably best to handle it separately in any case, an issue should be opened about it.

Reply:
Yeah, you are right. It might be too large change to do in this pull request, better done separately.